source/common/http/conn_pool_grid.cc

```diff
@@ -27,11 +27,16 @@ absl::string_view describePool(const ConnectionPool::Instance& pool) {
 static constexpr uint32_t kDefaultTimeoutMs = 300;
 
 std::string getSni(const Network::TransportSocketOptionsConstSharedPtr& options,
-                   Network::UpstreamTransportSocketFactory& transport_socket_factory) {
+                   Upstream::HostConstSharedPtr& host) {
   if (options && options->serverNameOverride().has_value()) {
     return options->serverNameOverride().value();
   }
-  return std::string(transport_socket_factory.defaultServerNameIndication());
+  std::string transport = std::string(host->transportSocketFactory().defaultServerNameIndication());
+  if (!transport.empty() ||
+      !Runtime::runtimeFeatureEnabled("envoy.reloadable_features.allow_alt_svc_for_ips")) {
+    return transport;
+  }
+  return host->hostname();
```

Comment:
Is this supposed to be an IP string?

Reply:
yeah, otherwise the SNI checks return the hostname

Reply:
Can you comment "if there is no configured SNI, try to return IP string as the host name"

Reply:
sure, done